File: 9.palindrome-number/nodejs/tests/palindrome_number.test.js
const { PalindromeNumber } = require('../src/palindrome_number')

describe('../src/palindrome_number.js', () => {
  it('positive number', () => {
    const inputs = {
      num: 121
    }
    const expected = true

    processTestCase(inputs, expected)
  })

  it('negative number', () => {
    const inputs = {
      num: -121
    }
    const expected = false

    processTestCase(inputs, expected)
  })

  it('zero ending number', () => {
    const inputs = {
      num: 10
    }
    const expected = false

    processTestCase(inputs, expected)
  })
})

const palindromeNumber = new PalindromeNumber()

const processTestCase = (inputs, expected) => {
  const actual = palindromeNumber.do(inputs.num)
  expect(actual).toBe(expected)
}

File: 9.palindrome-number/cpp/tests/palindrome_number_test.cpp
#include <gtest/gtest.h>
#include "../src/palindrome_number.h"

class Inputs {
public:
  int num;
};

void processTestCase(Inputs inputs, bool expected);

TEST(PalindromeNumberTest, PositiveNumber) {
  Inputs inputs {
    .num = 121
  };
  bool expected = true;

  processTestCase(inputs, expected);
}

TEST(PalindromeNumberTest, NegativeNumber) {
  Inputs inputs {
    .num = -121
  };
  bool expected = false;

  processTestCase(inputs, expected);
}

TEST(PalindromeNumberTest, ZeroEndingNumber) {
  Inputs inputs {
    .num = 10
  };
  bool expected = false;

  processTestCase(inputs, expected);
}

PalindromeNumber palindromeNumber;

void processTestCase(Inputs inputs, bool expected) {
  bool actual = palindromeNumber._do(inputs.num);
  ASSERT_EQ(actual, expected);
}
